modules/fedimint-ln-client/src/receive.rs

```diff
@@ -222,21 +222,34 @@ impl LightningReceiveConfirmedInvoice {
 
     async fn transition_funded(
         old_state: LightningReceiveStateMachine,
-        keypair: KeyPair,
+        receiving_key: ReceivingKey,
         result: Result<IncomingContractAccount, LightningReceiveError>,
         dbtx: &mut ClientSMDatabaseTransaction<'_, '_>,
         global_context: DynGlobalClientContext,
     ) -> LightningReceiveStateMachine {
         match result {
             Ok(contract) => {
-                let (txid, out_points) =
-                    Self::claim_incoming_contract(dbtx, contract, keypair, global_context).await;
-                LightningReceiveStateMachine {
-                    operation_id: old_state.operation_id,
-                    state: LightningReceiveStates::Funded(LightningReceiveFunded {
-                        txid,
-                        out_points,
-                    }),
+                match receiving_key {
+                    ReceivingKey::Internal(keypair) => {
+                        let (txid, out_points) =
+                            Self::claim_incoming_contract(dbtx, contract, keypair, global_context)
+                                .await;
+                        LightningReceiveStateMachine {
+                            operation_id: old_state.operation_id,
+                            state: LightningReceiveStates::Funded(LightningReceiveFunded {
+                                txid,
+                                out_points,
+                            }),
+                        }
+                    }
+                    ReceivingKey::External(_) => {
+                        // Claim successful
+                        // fixme is this okay?
+                        LightningReceiveStateMachine {
+                            operation_id: old_state.operation_id,
+                            state: LightningReceiveStates::Success(vec![]),
+                        }
+                    }
```

Comment:
Should these both cases even share one state machine? I think there's an easier way:

* Only spawn a tx submission SM for submitting the offer
* Don't spawn any lightning module-specific SM as there is nothing for the module to *do* beyond that point, we only want to *observe* payments
* In `subscribe_ln_receive_external` we first await acceptance of the offer submission tx and then wait for the contract being funded. (the two tiered state management is a bit complex, see e.g. how `subscribe_ln_receive` relates to the LN module receive state machine, happy to go into more detail if you want)